tuf/ngclient/_internal/metadata_bundle.py

```diff
@@ -0,0 +1,461 @@
+# Copyright the TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""TUF client bundle-of-metadata
+
+MetadataBundle keeps track of current valid set of metadata for the client,
```

Comment:
yeah we can, I've not been totally happy with the name. A lot of the code is indeed about implementing the update workflow so Workflow makes some sense... 

The core idea (or at least the big change) I was starting from is that the currently valid set of metadata must be considered a unit. Single metadata file cannot usually be considered "valid": validity can only be decided for the whole set -- and I was trying to figure out a name for this set (that could contain just root metadata or possibly dozens of individual metadata files)

Would be nice to get more opinions, I've been looking at this for way too long to be able to switch viewpoints now...

Reply:
Why no one likes my `TrustedMetadata` option :smile: 
This is how the specification refers to metadata files that have been verified by the client and persisted and `MetadataBundle` is implemented as a mapping class rather than a workflow.

Reply:
I like the name `TrusteMetadata`, but `MetadataBundle` is both a mapping and a workflow class, no?

Reply:
Yeah _Trusted_ would be nice in the name but I don't like TrustedMetadata because that sounds like a type of Metadata (a class we already have). The component does contain a lot of the update workflow, but in a way that workflow is just a side effect of keeping track of our collection of trusted metadata...